PCSX2/pcsx2
# Sio Phase 2.5 - Rewrite Pad Support + Add Guitar

### Description of Changes
- Completely replaces all pad code. 
- Reimplements pad protocol as discrete functions, uses an abstract PadBase with each actual controller type as a subclass, each with an independent implementation of the protocol matching that type of pad. 
- Added guitars. 
- GT4 should now properly engage DS2 mode again, so axis triggers can be used for throttle/brake controls again. 
- Other games which may have failed to engage DS2 mode should now be able to.

### Rationale behind Changes
Old pad code was soup, had issues with DS2 mode. 

### Suggested Testing Steps
- Verify inputs all function as expected. 
- Check stick inversion working. 
- Check using pressure (triggers on modern controllers are axes not buttons, bind these to PS2 face buttons).
- Check guitar games working.
- Check macros working.
- Check settings retention between game restarts and full emulator restarts.

### To-Do
- ~~Un-scuff the guitar mapping image.~~
- TAS is undoubtably broken.
- Probably more things.

### Casualties of War
- ~~PS1 mode is inoperable, SIO0 is disconnected from the pads for now.~~
